doc/cmis-module-enhancement/cmis-module-enhancement.md

```diff
@@ -0,0 +1,224 @@
+
+These initialization processes should be synchronized and the configuration of CMIS module should start only after the ASIC is initialized and per CMIS 5.2 it started sending the high-speed signal toward a module.
+
+Currently, SONIC uses the "host_tx_ready"  flag in the PORT table in STATE DB for synchronization. This flag is set by Ports OA right after the SAI API for setting the Admin status to UP returns with OK/Success status. PMON registers for changes of this flag in Redis DB and starts the CMIS initialization for a particular module when this flag is set.
+
+Current design has some gaps with synchronization between ASIC and module configuration.
+This document purpose is to introduce an enhacement that will address the gaps and find a backward compatible solution. 
+
+
+## 5. Requirements
+
+* SONiC shall have backward compatibility for platforms that don't support the proposed enhancement.
+
+* Vendor SDK/FW shall start sending high-speed signal to module, only when admin status is up, and it has an indication that module was plugged in.
+
+* Vendor SDK/FW shall support asynchronous notification of start/stop of sending high-speed signal from ASIC to module.
+
+* PortsOrch shall set host_tx_ready in state DB, only when it recieved notification that the high-speed signal is sent. 
+
+## 6. High-Level Design 
+
+Current Flow:
+<p align=left>
+<img src="img/The-Current-Flow.png" alt="Figure 1. Current Flow">
+</p>
+
+### 6.1. Problem Statement
+With move to SW-based management of CMIS modules - when the ASIC side is configured by vendor SDK, and module side by SONIC, some problems have been identified with the current approach.
+
+#### 6.1.2. Host TX Ready signal
+As mentioned earlier, currently SONIC assumes that as soon as the Admin Status is set to UP and the corresponding SAI call returns with OK/SUCCESS status, the "host_tx_ready" flag can be set in the STATE DB to trigger the CMIS State Machine for specific port. 
+But it is not always a truth. ASIC port initialization process takes some time and this time can increase with move to new transceiver technologies (e.g. BiDi). 
+So, in some cases the module initialization can be triggered too early, before the high-speed signal started to be transmitted by ASIC to a module
+
+#### 6.1.3. Unterminated Transmission
+With move to the SW-based module management, the module presence is handled by SONIC, and FW might be unaware of the module presence status. 
+In this case, when the Admin status of a port is set to UP, FW can start transmitting the high-speed signal even without a plugged-in module. Such unterminated transmission can cause cross-talks to adjacent ports, high EMI, high BER and eventually shorten the transceiver lifetime so it is recommended that ASIC will not start sending the high-speed signal before a module is plugged.
```

Comment:
I see a usecase where the some amount power is saved when the transmission is not enabled in ASIC.

Can you please elaborate on unterminated transmission ?   and how it cause cross-talks to adjacent ports?

Reply:
@jaganbal-a 
> I see a usecase where the some amount power is saved when the transmission is not enabled in ASIC.

The reason for this enhancement is not power. it is a side affect. 
The main reason here is to make sure the SerDes will not be damaged when high-speed signal will be sent without having a module plugged in to cage. 

> Can you please elaborate on unterminated transmission ? and how it cause cross-talks to adjacent ports?

Sending high-speed signal without having a plugged in module, can cause interruptions. for adjacent cables, it will cause more BER and again, can damage the SerDes.